AmplifyPlugins/API/AWSAPICategoryPluginIntegrationTests/GraphQL/GraphQLWithUserPoolIntegrationTests/AuthDirective/GraphQLAuthDirectiveIntegrationTests.swift

```diff
@@ -0,0 +1,244 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import Amplify
+import AWSMobileClient
+import AWSAPICategoryPlugin
+@testable import AWSAPICategoryPluginTestCommon
+@testable import AmplifyTestCommon
+import AWSPluginsCore
+
+class GraphQLAuthDirectiveIntegrationTests: XCTestCase {
+
+    static let amplifyConfiguration = "GraphQLAuthDirectiveIntegrationTests-amplifyconfiguration"
+    static let awsconfiguration = "GraphQLAuthDirectiveIntegrationTests-awsconfiguration"
+    static let credentials = "GraphQLAuthDirectiveIntegrationTests-credentials"
+    static var user1: String!
+    static var user2: String!
+    static var password: String!
+
+    static override func setUp() {
+        do {
+
+            let credentials = try TestConfigHelper.retrieveCredentials(
+                forResource: GraphQLAuthDirectiveIntegrationTests.credentials)
+
+            guard let user1 = credentials["user1"],
+                let user2 = credentials["user2"],
+                let password = credentials["password"] else {
+                    XCTFail("Missing credentials.json data")
+                    return
+            }
+
+            GraphQLAuthDirectiveIntegrationTests.user1 = user1
+            GraphQLAuthDirectiveIntegrationTests.user2 = user2
+            GraphQLAuthDirectiveIntegrationTests.password = password
+
+            let awsConfiguration = try TestConfigHelper.retrieveAWSConfiguration(
+                forResource: GraphQLAuthDirectiveIntegrationTests.awsconfiguration)
+            AWSInfo.configureDefaultAWSInfo(awsConfiguration)
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+    }
+
+    override func setUp() {
+        do {
+            AuthHelper.initializeMobileClient()
+
+            Amplify.reset()
+
+            try Amplify.add(plugin: AWSAPIPlugin())
+
+            let amplifyConfig = try TestConfigHelper.retrieveAmplifyConfiguration(
+                forResource: GraphQLAuthDirectiveIntegrationTests.amplifyConfiguration)
+            try Amplify.configure(amplifyConfig)
+
+            ModelRegistry.register(modelType: SocialNote.self)
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+    }
+
+    override func tearDown() {
+        Amplify.reset()
+        AuthHelper.signOut()
+    }
+
+    /// A model with operations [create, update, delete] allows the model to be read by others but not updated or
+    /// deleted. This creates a read-only model.
```

Comment:
I think we should adopt the following ascii table, and put something like this directly in our code.
```
Models created with:
   @auth(rules: [ { allow: owner, operations: [create, update, delete] } ])

Yield the following permissions model:
            Create   Read   Update   Delete
   Owner    allow   allow    allow    allow
   Others    x      allow    deny     deny

```

Reply:
thanks, adding this